internal/modulecontext/module_context.go

```diff
@@ -141,6 +144,19 @@ func (m ModuleContext) GetDatabase(name string, dbType DBType) (*sql.DB, error)
 	return db.db, nil
 }
 
+// LeaseClient is the interface for acquiring, heartbeating and releasing leases
+type LeaseClient interface {
+	// Returns ResourceExhausted if the lease is held.
+	Acquire(ctx context.Context, module string, key []string, ttl time.Duration) error
+	Heartbeat(ctx context.Context, module string, key []string, ttl time.Duration) error
+	Release(ctx context.Context, key []string) error
+}
+
+// MockLeaseClient provides a mock lease client when testing
+func (m ModuleContext) MockLeaseClient() optional.Option[LeaseClient] {
```

Comment:
Any more thought to my comment re. having an interface abstracting out the "real" and testing context?